### PR TITLE
#185 Must tolerate duplicate Events

### LINKED ISSUE #185: new guideline: must process events in a way that tolerates duplicates
This is a guideline for event consumers, not for the API itself.

Add some reasoning, like "at-least-once" delivery semantics.

### PR DESCRIPTION
**Pull request by maxim-tschumak**
New guideline: Must prepare for duplicate Events (closes #185).
As discussed in the last JF - this time more explanation and reasoning.

### CODE REVIEW

File: events/events.md [hunk 1]
```diff
@@ -150,4 +150,12 @@ Event types can follow these naming conventions (each convention has its own sho
  
  - Event type names must be url-safe. This is because the event type names are used by Nakadi as part of the URL for the event type and its stream.
 
- - Event type names should be lowercase words and numbers, using hypens, underscores or periods as separators. 
+ - Event type names should be lowercase words and numbers, using hypens, underscores or periods as separators.
```
Comment (fmueller) [inline events/events.md]: typo: hyphens

File: events/events.md [hunk 2]
```diff
@@ -150,4 +150,12 @@ Event types can follow these naming conventions (each convention has its own sho
...
+
+## {{ book.must }} Prepare for duplicate Events
+
+Event consumers must be able to process duplicate events.
+
+Most of the message broker and event buses, like Nakadi, guarantee “at-least-once” delivery. That is, one particular event is delivered to the consumer(s) one or more times. Other circumstances can also cause duplicate events.
```
Comment (fmueller) [inline events/events.md]: typo: brokers

Comment (fmueller) [inline events/events.md]: editorial: ...one particular event is delivered to consumers...

File: events/events.md [hunk 3]
```diff
@@ -150,4 +150,12 @@ Event types can follow these naming conventions (each convention has its own sho
...
+
+For example, described situation occurs if the publisher sends an event and cannot receive the acknowledgment (e.g. due to a network issue). In this case, the publisher will try to send the same event again. This leads to two identical events in the event bus which have to be processed by the consumer(s). Similar conditions can appear on the consumer side: an event has been processed successfully, but the consumer fails to confirm the processing.
```
Comment (fmueller) [inline events/events.md]: editorial: these situations occur ... and doesn't receive ... (due to network issues)

editorial: the consumer(s) -> consumers

eidtorial: on the consumer side -> on consumer side

**Comment by fmueller** [comment]
Good contribution, I only made editorial comments.

**Comment by maxim-tschumak** [comment]
Thanks for the comments @fmueller! I've included the proposed improvements.

**Comment by ePaul** [comment]
:+1:

**Comment by fmueller** [comment]
👍

**Pull request merged**